### PR TITLE
[Ceph] Add Integration Package with OSD Tree data stream

### PR DESCRIPTION
**Pull request by harnish-elastic**
- Enhancement

## What does this PR do?

- Added 1 data stream (OSD Tree metrics).
- Added data collection logic for the data streams.
- Added the ingest pipeline for the data streams.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yaml files.
- Added system test cases for the data stream.

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's changelog.yml file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/ceph directory.
- Run the following command to run tests.
elastic-package test

## Related issues

- Relates #4871
- Relates #4930
- Relates #4963
- Relates #5011 
- Relates #5070
- Relates #5128 
- Relates #5152

### CODE REVIEW

**Review by kush-elastic — APPROVED** [review]
LGTM! Approved

File: packages/ceph/data_stream/osd_tree/fields/fields.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        - name: depth
+          type: long
+          description: Depth of OSD node.
+        - name: device_class
+          type: keyword
+          description: The device class of OSD. i.e. hdd, ssd etc.
+        - name: exists
+          type: boolean
+          description: Represent OSD node still exist or not (1-true, 0-false).
+        - name: id
```
Comment (agithomas) [inline packages/ceph/data_stream/osd_tree/fields/fields.yml:23]: can we change this to osd_id or node_id which ever is most appropriate?

File: packages/ceph/data_stream/osd_tree/fields/fields.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          type: long
+          description: OSD or bucket node id.
+        - name: name
```
Comment (agithomas) [inline packages/ceph/data_stream/osd_tree/fields/fields.yml:26]: can we change this to osd_name or node_name whichever is most appropriate?

File: packages/ceph/data_stream/osd_tree/fields/fields.yml (hunk 1 above)
Comment (harnish-elastic) [reply packages/ceph/data_stream/osd_tree/fields/fields.yml:23]: This field contains both `osd_id` as well as `node_id`. We can change it to node_osd_id. Please let me know your suggestions.

File: packages/ceph/data_stream/osd_tree/fields/fields.yml (hunk 2 above)
Comment (harnish-elastic) [reply packages/ceph/data_stream/osd_tree/fields/fields.yml:26]: This field contains both `osd_name` as well as `node_name`. We can change it to node_osd_name. Please let me know your suggestions.

File: packages/ceph/data_stream/osd_tree/fields/fields.yml (hunk 1 above)
Comment (agithomas) [reply packages/ceph/data_stream/osd_tree/fields/fields.yml:23]: Lets go with node_osd_id

File: packages/ceph/data_stream/osd_tree/fields/fields.yml (hunk 2 above)
Comment (agithomas) [reply packages/ceph/data_stream/osd_tree/fields/fields.yml:26]: Lets go with node_osd_name

**Review by agithomas — APPROVED** [review]

**Pull request merged**